identity-account/src/stronghold/context.rs

```diff
@@ -78,49 +65,54 @@ async fn clear_expired_passwords() -> Result<()> {
   Ok(())
 }
 
-impl Context {
-  pub(crate) fn get() -> Result<&'static Self> {
-    static __THIS: OnceCell<Context> = OnceCell::new();
-    static __POOL: OnceCell<AsyncRuntime> = OnceCell::new();
-    static __SWEEP: Once = Once::new();
+static CONTEXT: Lazy<core::result::Result<Context, String>> = Lazy::new(|| {
+  let (sender, receiver) = std::sync::mpsc::channel();
 
-    let this: &'static Self = __THIS.get_or_try_init::<_, Error>(|| {
-      let system: ActorSystem = SystemBuilder::new()
-        // Disable the default actor system logger
-        .log(slog::Logger::root(slog::Discard, slog::o!()))
-        .create()?;
+  thread::spawn(move || {
+    let system_runner = System::new();
 
-      let stronghold: Stronghold = Stronghold::init_stronghold_system(system, Vec::new(), Vec::new());
-      let database: Arc<AsyncMutex<Database>> = Arc::new(AsyncMutex::new(Database::new(stronghold)));
+    let stronghold = system_runner
+      .block_on(Stronghold::init_stronghold_system(vec![], vec![]))
+      .map_err(|err| err.to_string());
```

Comment:
I wanted to do that. But the resulting `Lazy<Result<Context, Error>>` is then retrieved in the `Context::get` method. Since we only get a `&'static Result<Context, Error>` we cannot return an owned version of the error, nor `try!` it. Since the error is not `clonable` and cannot be made so easily, I figured this would be the easiest solution. I will add the todo for the anyhow change later.

In regards to the channels: they are used in a way where they never panic, so bubbling up the errors is not helpful for the caller anyway, imo.

Reply:
Are there any problems with deriving `Clone` for the `identity-account` `Error` to get around this problem, or is it due to something like `io::Error` not being `Clone`? 

I agree it's unclear whether `recv` can fail in such a way that the entire program doesn't panic, so it's probably fine not to return an error from it.

Reply:
Yeah, the `identity_account::error::Error` contains an `IoError` variant, so it is as you say :confused:.